### PR TITLE
ybt command of target info

### PR DESCRIPTION
**Pull request by dana-resonai**
Add a ybt command that prints information about the target as json. Currently remote_image_name, remote_image_tag, workspace.

Run example:
`ybt info registration/services:frame_recorder_service yio:cprint_pb ymath:ymath_types`
Output:
`{"registration/services:frame_recorder_service": {"workspace": "/Users/dana/work/resonai2/yabtwork/release_flavor/GoApp/registration_services_frame_recorder_service", "remote_image_name": "localhost:5000/yowza3d/frame_recorder_service", "remote_image_tag": "dca2c0eb290c3ffa0db94a6a4f1c8fab3596c989"}, "yio:cprint_pb": {"workspace": "/Users/dana/work/resonai2/yabtwork/release_flavor/CppApp/yio_cprint_pb", "remote_image_name": null, "remote_image_tag": null}, "ymath:ymath_types": {"workspace": "/Users/dana/work/resonai2/yabtwork/release_flavor/CppLib/ymath_ymath_types"}}`

### CODE REVIEW

File: yabt/target_info.py [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+"""
+
+import json
+
+from .config import Config
+from .logging import make_logger
+from .target_utils import parse_target_selectors
+
+
+logger = make_logger(__name__)
```
Comment (itamaro) [inline yabt/target_info.py:31]: remove if unused

File: yabt/target_info.py [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+  targets = parse_target_selectors(conf.targets, conf)
+  targets_info = {}
+  for target_name in targets:
+    target = build_context.targets[target_name]
+    info = {
+      'workspace': build_context.get_workspace(target.builder_name, target_name)
+    }
+    if 'image_caching_behavior' in target.props:
+      info['remote_image_name'] = target.props.image_caching_behavior.get('remote_image_name', None)
+      info['remote_image_tag'] = target.props.image_caching_behavior.get('remote_image_tag', None)
```
Comment (itamaro) [inline yabt/target_info.py:44]: is it useful to have `null` in the output when these are not defined? wouldn't it be cleaner to not include them if they are `None`s?

**Review by itamaro — COMMENTED** [review]
can you explain the use-case for this?

is it for machine processing? or for humans?

if for humans - why json? (vs some nicely formatted table or something)
if for machines - what for?

**Comment by dana-resonai** [comment]
> can you explain the use-case for this?
> 
> is it for machine processing? or for humans?
> 
> if for humans - why json? (vs some nicely formatted table or something) if for machines - what for?

It is for machines. The idea is to use it in ci-cd, in our case - we sometimes want to push the image we just built to some docker repository, and fir this we need to know the image name.
The workspace is for a similar case when we want to find some files ybt created for this target.

File: yabt/target_info.py (hunk 2 above)
Comment (dana-resonai) [reply yabt/target_info.py]: Fixed it. The example is now:
`{"registration/services:frame_recorder_service": {"workspace": "/Users/dana/work/resonai2/yabtwork/release_flavor/GoApp/registration_services_frame_recorder_service", "remote_image_name": "localhost:5000/yowza3d/frame_recorder_service", "remote_image_tag": "0e906765a932367a3c6133b373d86d4b6ac6b2cd"}, "yio:cprint_pb": {"workspace": "/Users/dana/work/resonai2/yabtwork/release_flavor/CppApp/yio_cprint_pb"}, "ymath:ymath_types": {"workspace": "/Users/dana/work/resonai2/yabtwork/release_flavor/CppLib/ymath_ymath_types"}}`

Comment (itamaro) [reply yabt/target_info.py]: thanks, consider adding a test for this :-)

**Comment by itamaro** [comment]
> It is for machines. The idea is to use it in ci-cd, in our case - we sometimes want to push the image we just built to some docker repository, and fir this we need to know the image name.

hmm, I thought there is already a way to do this using the `artifacts.json` file - is the goal here different / not supported using artiftacts.json (or by extending artifacts.json)?

**Comment by dana-resonai** [comment]
> hmm, I thought there is already a way to do this using the `artifacts.json` file - is the goal here different / not supported using artiftacts.json (or by extending artifacts.json)?

I don't understand how to use `artifacts.json` for it. It doesn't have any of the data I need, and also you will need to run some ybt code in order to calculate the target's hash to find the `artifacts.json` directory. Can you explain more?

**Comment by itamaro** [comment]
< 

> I don't understand how to use `artifacts.json` for it. It doesn't have any of the data I need, and also you will need to run some ybt code in order to calculate the target's hash to find the `artifacts.json` directory. Can you explain more?

I'm not 100% on the details :-) I think artifacts.json was added as an output after a build command to help CI/CD find the images that were built, so it definitely depends on running a build.

if it's not doing what you need, this lgtm. thanks for adding a test!

btw, what's up with travis tests?

**Review by itamaro — APPROVED** [review]

**Comment by dana-resonai** [comment]
> btw, what's up with travis tests?
Not sure, we are trying to figure out. There were many environment problems which @eyal-resonai fixed, now there is a test that fails constantly in travis but passes locally.

**Pull request merged**